maps-compose/src/main/java/com/google/maps/android/compose/streetview/StreetView.kt

```diff
@@ -0,0 +1,145 @@
+package com.google.maps.android.compose.streetview
+
+import android.content.ComponentCallbacks
+import android.content.res.Configuration
+import android.os.Bundle
+import android.util.Log
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.Composition
+import androidx.compose.runtime.CompositionContext
+import androidx.compose.runtime.DisposableEffect
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.MutableState
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.rememberCompositionContext
+import androidx.compose.runtime.rememberUpdatedState
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.platform.LocalLifecycleOwner
+import androidx.compose.ui.viewinterop.AndroidView
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.LifecycleEventObserver
+import com.google.android.gms.maps.StreetViewPanoramaOptions
+import com.google.android.gms.maps.StreetViewPanoramaView
+import com.google.android.gms.maps.model.StreetViewPanoramaCamera
+import com.google.android.gms.maps.model.StreetViewPanoramaOrientation
+import com.google.maps.android.compose.disposingComposition
+import com.google.maps.android.ktx.MapsExperimentalFeature
+import com.google.maps.android.ktx.awaitStreetViewPanorama
+
+@MapsExperimentalFeature
+@Composable
+public fun StreetView(
+    modifier: Modifier = Modifier,
+    cameraPositionState: StreetViewCameraPositionState = rememberStreetViewCameraPositionState(),
+    streetViewPanoramaOptionsFactory: () -> StreetViewPanoramaOptions = {
+        StreetViewPanoramaOptions()
+    },
+    isPanningGesturesEnabled: Boolean = true,
+    isStreetNamesEnabled: Boolean = true,
+    isUserNavigationEnabled: Boolean = true,
+    isZoomGesturesEnabled: Boolean = true,
+    onClick: (StreetViewPanoramaOrientation) -> Unit = {},
+    onLongClick: (StreetViewPanoramaOrientation) -> Unit = {},
+) {
+    val context = LocalContext.current
+    val streetView =
+        remember { StreetViewPanoramaView(context, streetViewPanoramaOptionsFactory()) }
```

Comment:
How come the panorama options can only be set once? Something preventing them from being updated mid-use?

Reply:
Unfortunately this is a restriction with `StreetViewPanoramaView`, an options object can only be provided at construction time. The same pattern is observed for `GoogleMapsOptions` https://developers.google.com/android/reference/com/google/android/gms/maps/StreetViewPanoramaView#StreetViewPanoramaView(android.content.Context,%20com.google.android.gms.maps.StreetViewPanoramaOptions)

Reply:
Ah I see, lame